src/rubrix/server/apis/v0/handlers/text_classification.py

```diff
@@ -238,17 +239,20 @@ def grouper(n, iterable, fillvalue=None):
         if limit:
             stream = takeuntil(stream, limit=limit)
 
-        for batch in grouper(
-            n=chunk_size,
-            iterable=stream,
-        ):
-            filtered_records = filter(lambda r: r is not None, batch)
-            yield "\n".join(
-                map(
-                    lambda r: r.json(by_alias=True, exclude_none=True), filtered_records
-                )
-            ) + "\n"
-
+        try:
+            for batch in grouper(
+                    n=chunk_size,
+                    iterable=stream,
+            ):
+                filtered_records = filter(lambda r: r is not None, batch)
+                yield "\n".join(
+                    map(
+                        lambda r: r.json(by_alias=True, exclude_none=True), filtered_records
+                    )
+                ) + "\n"
+        except Exception as error:
+            yield errors.exception_to_rubrix_error(error)
+            return
     return StreamingResponse(
```

Comment:
we can revert all the changes if we use the `StreamingResponseWIthErrorHandling` instead of the `StreamingResponse` class